llvm/include/llvm/CodeGen/MachineFrameInfo.h

```diff
@@ -282,6 +284,10 @@ class MachineFrameInfo {
   /// It is only valid during and after prolog/epilog code insertion.
   uint64_t MaxCallFrameSize = ~UINT64_C(0);
 
+  /// Call frame sizes for the MachineFunction's MachineBasicBlocks. This is set
+  /// by the MachineFrameSizeInfo constructor and cleared by its destructor.
+  MachineFrameSizeInfo *SizeInfo = nullptr;
```

Comment:
I don't think this type of ownership makes sense. MachineFrameInfo should be a compilation wide state as the function is processed. This shouldn't be set for one function in one pass that happens to compute it

Reply:
The motivation for that was to make passing this info to `X86TargetLowering::EmitLoweredTLSAddr` less obtrusive in #106965. For this PR here, it's certainly not necessary. I can move this change over to the other PR for now and then we can discuss there how else to pass it to the function (or decide on a different way to avoid the nested CALLSEQs for TLSAddrs).